sim/riscv/sim-main.c

```diff
@@ -1678,11 +2146,26 @@ void step_once (SIM_CPU *cpu)
     sim_engine_halt (sd, cpu, NULL, pc, sim_signalled, SIM_SIGILL);
 
   for (; op->name; op++)
-    if ((op->match_func) (op, iw) && !(op->pinfo & INSN_ALIAS))
-      {
-	pc = execute_one (cpu, iw, op);
-	break;
-      }
+    {
+      /* Does the opcode match?  */
+      if (!(op->match_func) (op, iw))
+	continue;
+      /* Is this a pseudo-instruction?  */
+      if ((op->pinfo & INSN_ALIAS))
+	continue;
+      /* Is this instruction restricted to a certain value of XLEN?  */
+      if (isdigit (op->subset[0]) && atoi (op->subset) != xlen)
+	continue;
+      /* It's a match.  */
+      pc = execute_one (cpu, iw, op);
+      break;
+
+      if ((op->match_func) (op, iw) && !(op->pinfo & INSN_ALIAS))
```

Comment:
Dead code after a break that should be removed.  This is the original code which is no longer useful.